src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -842,7 +841,8 @@ private LanguageExpression ConvertVariableAccess(VariableAccessSyntax variableAc
                     return GetLocalVariableExpression(localVariableSymbol);
 
                 default:
-                    throw new NotImplementedException($"Encountered an unexpected symbol kind '{symbol?.Kind}' when generating a variable access expression.");
+                    //just return a call to variables.
+                    return CreateFunction("variables", new JTokenExpression(name));                    
```

Comment:
>CreateFunction [](http://example.com/codeflow?start=27&length=14)

default case should continue to throw, so it blows up when a new symbol is added.

Reply:
I created a special class called `ExplicitVariableAcccessSyntax` which is being detected in the ExpressionConverter and symbol check is being skipped for that C# type - we will always return `variable()` call for it.